neural_lam/models/ar_model.py

```diff
@@ -21,35 +22,67 @@ class ARModel(pl.LightningModule):
     # pylint: disable=arguments-differ
     # Disable to override args/kwargs from superclass
 
-    def __init__(self, args):
+    def __init__(
+        self, args, datastore: BaseDatastore, forcing_window_size: int
+    ):
         super().__init__()
-        self.save_hyperparameters()
+        self.save_hyperparameters(ignore=["datastore"])
         self.args = args
-        self.config_loader = config.Config.from_file(args.data_config)
-
-        # Load static features for grid/data
-        static_data_dict = utils.load_static_data(
-            self.config_loader.dataset.name
+        self._datastore = datastore
+        # XXX: should be this be somewhere else?
+        split = "train"
```

Comment:
Yes, `ARModel` is not instantiated per split, so this seems a bit strange. This is only used to get the static features, but the static features are per definition the same for all splits. So a better way to handle this would be to allow for retrieving static features from a datastore without having to specify a split, or just allow for split to be set to `None` when getting static features. A clarifying comment here would probably also be good.

Reply:
I have implemented a test which checks that static data can be loaded with `split=None` in https://github.com/leifdenby/neural-lam/commit/a41c314f548d269a09ba5759beb820ff2c3bc212